### PR TITLE
nxp: imx8ulp: add support for and enable pinctrl

### PR DESCRIPTION
**Pull request by LaurentiuM1234**
Depends on:

- [x] https://github.com/zephyrproject-rtos/zephyr/pull/70219

### CODE REVIEW

**Review by iuliana-prodan — DISMISSED** [review]

**Review by dbaluta — DISMISSED** [review]

**Comment by LaurentiuM1234** [comment]
@gmarull pls review

File: dts/bindings/pinctrl/nxp,imx8ulp-pinctrl.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      drive-strength:
+        type: boolean
+        description: |
+          Use this property to enable drive strength. If specified,
+          this property will be used to configure the pad for high
+          drive strength. If not present, the standard drive strength
+          will be used.
+      slew-rate:
+        type: boolean
+        description: |
```
Comment (gmarull) [inline dts/bindings/pinctrl/nxp,imx8ulp-pinctrl.yaml:33]: this property is misleading as a boolean, please, make it a choice between standard/slow (also, explain what "standard" means...)

**Review by gmarull — CHANGES_REQUESTED** [review]

File: dts/bindings/pinctrl/nxp,imx8ulp-pinctrl.yaml (hunk 1 above)
Comment (LaurentiuM1234) [reply dts/bindings/pinctrl/nxp,imx8ulp-pinctrl.yaml]: Thanks! Switched to `string` type + enum. Also changed comments for `slew-rate` and `drive-strength`. Should be more useful now.

**Review by gmarull — DISMISSED** [review]

**Comment by LaurentiuM1234** [comment]
**V2 changes**

1. Changed type of `slew-rate` and `drive-strength` properties to `string`. Also, modified comments to make the properties more clear.
2. Rebased.

**Review by iuliana-prodan — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**